File: wheater_script.py
import requests

def get_weather(city, api_key):
    base_url = "http://api.openweathermap.org/data/2.5/weather"
    params = {
        "q": city,
        "appid": api_key,
        "units": "metric"  # Cambia a "imperial" para Fahrenheit
    }

    try:
        response = requests.get(base_url, params=params)
        response.raise_for_status()  # Lanza una excepción si la respuesta tiene un error HTTP
        data = response.json()

        # Extraer información relevante
        city_name = data["name"]
        temperature = data["main"]["temp"]
        weather_description = data["weather"][0]["description"]

        print(f"Weather in {city_name}:")
        print(f"Temperature: {temperature}°C")
        print(f"Description: {weather_description.capitalize()}")

    except requests.exceptions.RequestException as e:
        print(f"Error fetching weather data: {e}")
    except KeyError:
        print("Error: Unable to parse weather data. Please check the city name or API key.")

if __name__ == "__main__":
    # Reemplaza con tu clave de API de OpenWeatherMap
    API_KEY = "f14a0ca5c7a31440e709d1677124c38e"
    city = input("London")
    get_weather(city, API_KEY)

File: wheater_script.js
async function getWeather(city, apiKey) {
    const baseUrl = "http://api.openweathermap.org/data/2.5/weather";
    const url = `${baseUrl}?q=${city}&appid=${apiKey}&units=metric`; // Cambia "metric" a "imperial" para Fahrenheit

    try {
        const response = await fetch(url);
        if (!response.ok) {
            throw new Error(`HTTP error! status: ${response.status}`);
        }

        const data = await response.json();

        // Extraer información relevante
        const cityName = data.name;
        const temperature = data.main.temp;
        const weatherDescription = data.weather[0].description;

        console.log(`Weather in ${cityName}:`);
        console.log(`Temperature: ${temperature}°C`);
        console.log(`Description: ${weatherDescription.charAt(0).toUpperCase() + weatherDescription.slice(1)}`);
    } catch (error) {
        console.error(`Error fetching weather data: ${error.message}`);
    }
}

// Reemplaza con tu clave de API de OpenWeatherMap
const API_KEY = "f14a0ca5c7a31440e709d1677124c38e";
const city = "London";
getWeather(city, API_KEY);